src/Data/Fin/Induction.agda

```diff
@@ -87,3 +98,56 @@ module _ {ℓ} where
     ; wfRec        to ≺-rec
     )
     hiding (wfRec-builder)
+
+
+------------------------------------------------------------------------
+-- Well-foundedness of other (strict) partial orders on Fin
+
+module _ {n e} {_≈_ : Rel (Fin n) e} where
+
+  -- Every (strict) partial order over `Fin n' is well-founded.
+
+  -- Intuition: there cannot be any infinite descending chains simply
+  -- because Fin n has only finitely many inhabitants.  Thus any chain
+  -- of length > n must have a cycle (which is forbidden by
+  -- irreflexivity).
+
+  spo-wellFounded : ∀ {r} {_⊏_ : Rel (Fin n) r} →
+                    IsStrictPartialOrder _≈_ _⊏_ → WellFounded _⊏_
+  spo-wellFounded {_} {_⊏_} isSPO i = go n i pigeon where
+
+    module ⊏ = IsStrictPartialOrder isSPO
+
+    go : ∀ m i →
+         ((xs : Vec (Fin n) m) → Linked (flip _⊏_) (i ∷ xs) → WellFounded _⊏_) →
+         Acc _⊏_ i
+    go zero    i k = k [] [-] i
+    go (suc m) i k = acc $ λ j j⊏i → go m j (λ xs i∷xs↑ → k (j ∷ xs) (j⊏i ∷ i∷xs↑))
+
+    pigeon : (xs : Vec (Fin n) n) → Linked (flip _⊏_) (i ∷ xs) → WellFounded _⊏_
+    pigeon xs i∷xs↑ =
+      let (i₁ , i₂ , i₁<i₂ , xs[i₁]≡xs[i₂]) = pigeonhole (n<1+n n) (Vec.lookup (i ∷ xs)) in
+      let xs[i₁]⊏xs[i₂] = Linkedₚ.lookup (Flip.transitive _⊏_ ⊏.trans) {xs = i ∷ xs} i₁<i₂ i∷xs↑ in
+      let xs[i₁]⊏xs[i₁] = ⊏.<-respʳ-≈ (⊏.Eq.reflexive xs[i₁]≡xs[i₂]) xs[i₁]⊏xs[i₂] in
+      contradiction xs[i₁]⊏xs[i₁] (⊏.irrefl ⊏.Eq.refl)
+
+  po-wellFounded : ∀ {r} {_⊑_ : Rel (Fin n) r} →
+                   IsPartialOrder _≈_ _⊑_ → WellFounded (ToStrict._<_ _≈_ _⊑_)
+  po-wellFounded isPO =
+    spo-wellFounded (ToStrict.<-isStrictPartialOrder _≈_ _ isPO)
+
+  -- The inverse order is also well-founded, i.e. every (strict)
+  -- partial order is also Noetherian.
+
+  spo-noetherian : ∀ {r} {_⊏_ : Rel (Fin n) r} →
+                   IsStrictPartialOrder _≈_ _⊏_ → WellFounded (flip _⊏_)
+  spo-noetherian isSPO = spo-wellFounded >-isStrictPartialOrder
+    where
+      spo : StrictPartialOrder _ _ _
+      spo = record { isStrictPartialOrder = isSPO }
+      open SPO spo using (_>_; >-isStrictPartialOrder)
```

Comment:
We could avoid all the opening and closing of records if you went directly via `Relation.Binary.Construct.Converse`?

https://github.com/agda/agda-stdlib/blob/eb89d28ac6c12eaa15cce6f9ca707c247273f6da/src/Relation/Binary/Construct/Converse.agda#L130-L131

Reply:
Makes sense. I wasn't aware of that module. But then I'm wondering whether the changes to src/Relation/Binary/Properties/StrictPartialOrder.agda actually make sense. It seems like they're just duplicates. Maybe I should remove them from the PR, WDYT?

Reply:
OK, as a compromise, I left the new proofs but refactored them to use `Converse`. If you feel that the new stuff is superfluous, I don't mind reverting the changes altogether.

Reply:
So yes in some sense they are just duplicates, but I still think they're valuable as they're optimised for different dimensions. `Construct.Converse` is for flipping multiple structures, whereas the idea of the `Relation.Properties.X` module is that you can pass in a specific structure `X` and get all the relation constructions for that structure with meaningful names instantly in scope.